src/Dibi/Drivers/FirebirdDriver.php

```diff
@@ -30,7 +30,7 @@ class FirebirdDriver implements Dibi\Driver
 
 	public const ERROR_EXCEPTION_THROWN = -836;
 
-	/** @var resource */
+	/** @var resource|null */
```

Comment:
is never null

Reply:
I think `$connection` can be null, because of this line:

![Snímek obrazovky 2020-05-07 v 22 31 07](https://user-images.githubusercontent.com/4738758/81341808-733a4b00-90b2-11ea-869a-8f518f4dcff8.png)

The default value is defined in `if` statement in `__construct` which can be ignored.

Reply:
It must be never null, is_resource is tricky https://3v4l.org/2ps6c

Reply:
I am not sure I understand. Your function `is_null()` return `false` so type is not null:

![Snímek obrazovky 2020-05-07 v 22 51 30](https://user-images.githubusercontent.com/4738758/81343481-5bb09180-90b5-11ea-87f6-04c4fe91b127.png)

Reply:
I reverted. Thanks.